src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -192,6 +192,11 @@ public static BinaryData FromObjectAsJson<T>(T jsonSerializable, JsonSerializerO
         /// <summary>
         /// Converts the value of this instance to a string using UTF-8.
         /// </summary>
+        /// <remarks>
+        /// If the underlying object is a JSON string, calling <see cref="ToString"/> will retain the wrapping double quotes in the
+        /// resulting string. If you want to deserialize the JSON string to a string, thereby removing the wrapping double quotes,
+        /// call <see cref="ToObjectFromJson{String}"/> instead.
```

Comment:
Sharing Levi's "this feels too special-cased", I think something more like

```suggestion
        /// No special treatment is given to the contents of the data, it is merely decoded as a UTF-8 string.
        /// For a JPEG or other binary file format the string will largely be nonsense with many embedded NUL characters,
        /// and UTF-8 JSON values will look like their file/network representation,
        /// including starting and stopping quotes on a string.
```

And then add a `<seealso cref="ToObjectFromJson{String}" />` after the `</returns>`